index.es6

```diff
@@ -1,26 +1,38 @@
-export default function trimCanvas (canvas) {
-  let context = canvas.getContext('2d')
+export default function trimCanvas (canvas, options = {}) {
+  const {clone} = options
+  let {createCanvas = createCanvasDOM} = options // support node-canvas, etc
 
-  let imgWidth = canvas.width
-  let imgHeight = canvas.height
+  const context = canvas.getContext('2d')
 
-  let imgData = context.getImageData(0, 0, imgWidth, imgHeight).data
+  const imgWidth = canvas.width
+  const imgHeight = canvas.height
+
+  const imgData = context.getImageData(0, 0, imgWidth, imgHeight).data
 
   // get the corners of the relevant content (everything that's not white)
-  let cropTop = scanY(true, imgWidth, imgHeight, imgData)
-  let cropBottom = scanY(false, imgWidth, imgHeight, imgData)
-  let cropLeft = scanX(true, imgWidth, imgHeight, imgData)
-  let cropRight = scanX(false, imgWidth, imgHeight, imgData)
+  const cropTop = scanY(true, imgWidth, imgHeight, imgData)
+  const cropBottom = scanY(false, imgWidth, imgHeight, imgData)
+  const cropLeft = scanX(true, imgWidth, imgHeight, imgData)
+  const cropRight = scanX(false, imgWidth, imgHeight, imgData)
 
   // + 1 is needed because this is a difference, there are n + 1 pixels in
   // between the two numbers inclusive
-  let cropXDiff = (cropRight - cropLeft) + 1
-  let cropYDiff = (cropBottom - cropTop) + 1
+  const cropXDiff = (cropRight - cropLeft) + 1
+  const cropYDiff = (cropBottom - cropTop) + 1
 
   // get the relevant data from the calculated coordinates
-  let trimmedData = context.getImageData(cropLeft, cropTop, cropXDiff,
+  const trimmedData = context.getImageData(cropLeft, cropTop, cropXDiff,
     cropYDiff)
 
+  // early return if cloning
+  if (clone) {
```

Comment:
Per https://github.com/agilgur5/trim-canvas/pull/9#discussion_r352382366, it probably makes more sense to clone at the top of the function, then the rest of the logic can run uninterrupted on the clone. That would also replicate how the library is used right now without this feature -- one clones the canvas, then calls `trimCanvas` on the clone.

Reply:
The downside is that that method is probably more inefficient, since we'd be copying data into the clone twice, one for cloning and one for trimming. This way it only occurs once, at the expense of somewhat more complicated logic (also that might've been why I did it this way in the first place, but idr). Hmm......